src/file_storage.cpp

```diff
@@ -792,71 +792,73 @@ namespace libtorrent
 		return ret;
 	}
 
-	string_view file_storage::file_name(int index) const
+	string_view file_storage::file_name(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		internal_file_entry const& fe = m_files[index];
+		internal_file_entry const& fe = m_files[std::size_t(index)];
 		return fe.filename();
 	}
 
-	std::int64_t file_storage::file_size(int index) const
+	std::int64_t file_storage::file_size(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return m_files[index].size;
+		return std::int64_t(m_files[std::size_t(index)].size);
 	}
 
-	bool file_storage::pad_file_at(int index) const
+	bool file_storage::pad_file_at(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return m_files[index].pad_file;
+		return m_files[std::size_t(index)].pad_file;
 	}
 
-	std::int64_t file_storage::file_offset(int index) const
+	std::int64_t file_storage::file_offset(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return m_files[index].offset;
+		return std::int64_t(m_files[std::size_t(index)].offset);
```

Comment:
is the offset field unsigned here? it's not clear why you need this cast. would it be better to make the offset field signed?